cpp/src/arrow/compute/api_scalar.h

```diff
@@ -346,6 +347,23 @@ class ARROW_EXPORT SliceOptions : public FunctionOptions {
   int64_t start, stop, step;
 };
 
+class ARROW_EXPORT ListSliceOptions : public FunctionOptions {
+ public:
+  explicit ListSliceOptions(int64_t start, std::optional<int64_t> stop = std::nullopt,
+                            int64_t step = 1, bool return_fixed_size_list = true);
+  ListSliceOptions();
+  static constexpr char const kTypeName[] = "ListSliceOptions";
+  /// The start of list slicing.
+  int64_t start;
+  /// Optional stop of list slicing. If not set, then slice to end. (NotImplemented)
+  std::optional<int64_t> stop;
+  /// Slicing step
+  int64_t step;
+  /// Whether to return a FixedSizeListArray. If stop is after a list element's length,
+  /// nulls will be appended to the requested slice size.
```

Comment:
the "appending nulls" is only in case of FixedSizeList output? We should mention somewhere that this behaviour for variable size list is different